Hotfix/map my finals

ecbrady opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/byu-oit-appdev/ios-byusuite/50)
<!-- Reviewable:end -->

romrell4 commented:
Reviewed 15 of 15 files at r1.
Review status: all files reviewed at latest revision, 19 unresolved discussions.

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 28 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KYtz1A-KBhEGbKsf2Uf:-KYtz1A-KBhEGbKsf2Ug:b3efu1g) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/00ef8982c080084c45c6937211f997678026ca6f/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L28)):*
> ```objc
> @property (nonatomic) SemesterButton *semButton;
> @property (nonatomic) RegYearTerm *yearTerm;
> @property (nonatomic) ByuFinalsScheduleRequest *finalsRequest;
> ```

We need to replace this with a Client. Maybe that wasn't part of this phase, but we'll have to do it before we're done with this feature

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 29 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KYtz6VChBJRbXQJoBeR:-KYtz6VChBJRbXQJoBeS:buv2suq) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/00ef8982c080084c45c6937211f997678026ca6f/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L29)):*
> ```objc
> @property (nonatomic) RegYearTerm *yearTerm;
> @property (nonatomic) ByuFinalsScheduleRequest *finalsRequest;
> @property (nonatomic) BOOL finalsLoaded;
> ```

Can't this just be known if finals != nil?

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 34 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KYtzCn0q8y_o8z80Gcw:-KYtzCn0q8y_o8z80Gcx:b-33b6k6) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/00ef8982c080084c45c6937211f997678026ca6f/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L34)):*
> ```objc
> @property (nonatomic) NSMutableArray<ByuFinal *> *unscheduledFinals;
> @property (nonatomic) NSArray<CampusBuilding *> *buildings;
> @property (nonatomic) BOOL buildingsLoaded;
> ```

See comment above

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 54 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KYtzW3LWq36H2SkTcJy:-KYtzW3LWq36H2SkTcJz:bp70cwv) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/00ef8982c080084c45c6937211f997678026ca6f/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L54)):*
> ```objc
>     [super viewDidAppear:animated];
>     
>     if (self.semButton) {
> ```

What is this if doing? Is this if we come back to it?

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 61 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KYtzc-wHziPwaMLJl86:-KYtzc-wHziPwaMLJl87:b3d09ha) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/00ef8982c080084c45c6937211f997678026ca6f/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L61)):*
> ```objc
> -(void)dealloc {
>     self.finalsRequest.delegate = nil;
>     self.tabBarController.delegate = nil;
> ```

We shouldn't need the second one. The first one will go away when we use a Client

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 76 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KYtzwMrXaeIAOtGZ6gb:-KYtzwMrXaeIAOtGZ6gc:b5fp44q) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/00ef8982c080084c45c6937211f997678026ca6f/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L76)):*
> ```objc
>         return YES;
>     } else {
>         return NO;
> ```

Put a comment here about why you're not letting them switch tabs

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 83 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KYu--wTGRsYL6UyOswA:-KYu--wTGRsYL6UyOswB:baoqb77) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/00ef8982c080084c45c6937211f997678026ca6f/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L83)):*
> ```objc
> 
> -(NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
>     if (self.unscheduledFinals.count == 0) {
> ```

Use ternary here

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 104 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KYu-RUh-QmoM9tiZ3Ls:-KYu-RUh-QmoM9tiZ3Lt:bf8tr01) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/00ef8982c080084c45c6937211f997678026ca6f/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L104)):*
> ```objc
>                                      ((final.classPeriod == nil) ? @"" : final.classPeriod)];
>     } else if (indexPath.section == 1) {
>         cell = [self.tableView dequeueReusableCellWithIdentifier:@"unscheduledFinalCell"];
> ```

Are the two of these more different that just a nil detailsTextLabel?

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 124 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KYu-g11zPOyGUnOQjKF:-KYu-g11zPOyGUnOQjKG:b3gdbmj) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/00ef8982c080084c45c6937211f997678026ca6f/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L124)):*
> ```objc
> }
> 
> -(NSString *)tableView:(UITableView *)tableView titleForHeaderInSection:(NSInteger)section {
> ```

Let's move this up next to the number of sections

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 125 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KYu-v9_eSoRsXtIXEGe:-KYu-v9_eSoRsXtIXEGf:b-rl59mq) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/00ef8982c080084c45c6937211f997678026ca6f/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L125)):*
> ```objc
> 
> -(NSString *)tableView:(UITableView *)tableView titleForHeaderInSection:(NSInteger)section {
>     if (section == 0 && self.finals.count != 0) {
> ```

I don't think this logic is correct... If self.finals.count == 0, there will still be two sections from what it looks like....

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 129 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KYu03n6cgZXe6alER1W:-KYu03n6cgZXe6alER1X:bdusagp) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/00ef8982c080084c45c6937211f997678026ca6f/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L129)):*
> ```objc
>     } else if (section == 1) {
>         return @"  Unscheduled Finals";
>     } else {
> ```

If you only have two, you probably don't need the else if. Just use an if/else

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 135 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KYu0EPtsDJw7q-NpDyW:-KYu0EPtsDJw7q-NpDyX:b-c0x8a3) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/00ef8982c080084c45c6937211f997678026ca6f/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L135)):*
> ```objc
> 
> -(void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
>     [self.tableView deselectRowAtIndexPath:indexPath animated:YES];
> ```

If we can do this when the view returns, that would be pretty cool :)

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 222 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KYu0qUnZ8t0tlm-jQPq:-KYu0qUonATV-oG2Ebjp:b-3sm390) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/00ef8982c080084c45c6937211f997678026ca6f/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L222)):*
> ```objc
> }
> 
> -(void)loadDataIfReady {
> ```

This actually seems more like a 

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 225 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KYu3mscVwi2N6o75i67:-KYu3msdmMkl8K8VWbs3:b-yztjoh) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/00ef8982c080084c45c6937211f997678026ca6f/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L225)):*
> ```objc
>     if (self.buildingsLoaded && self.finalsLoaded) {
>         self.buildingsLoaded = NO;
>         self.finalsLoaded = NO;
> ```

Is this loading the buildings and final everytime the tab is selected? I don't think that would be desirable

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 233 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KYu40Bxj63xPvYvKmW7:-KYu40Bxj63xPvYvKmW8:b-2i69gt) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/00ef8982c080084c45c6937211f997678026ca6f/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L233)):*
> ```objc
>             self.tableFooter.hidden = NO;
>         }
>         [self scheduledAndUnscheduledSeparator:self.finals];
> ```

Should this be called [separateIntoScheduledAndUnscheduledFinals:]?

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsMapViewController.m, line 44 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KYu57UX3c6GIDPaS4qU:-KYu57UX3c6GIDPaS4qV:bfzxipc) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/00ef8982c080084c45c6937211f997678026ca6f/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsMapViewController.m#L44)):*
> ```objc
>         CampusBuilding *selectedBuilding = nil;
>         
>         for (id<MKAnnotation> annotation in self.mapView.annotations) {
> ```

I think we wrote a higherOrder function to do this. Something called findByBlock or something like that. Try to find that in NSArray+higherOrder.h, and if you have questions, ask me

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsMapViewController.m, line 52 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KYu5Ks5uGPjTcC2FyAX:-KYu5Ks5uGPjTcC2FyAY:bqtem89) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/00ef8982c080084c45c6937211f997678026ca6f/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsMapViewController.m#L52)):*
> ```objc
>             
>             if ([[currentBuilding.building lowercaseString] isEqualToString:[self.selectedFinal.building lowercaseString]]) {
>                 selectedBuilding = currentBuilding;
> ```

If we can't use higher order, we should at least put a break here

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsMapViewController.m, line 125 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KYu7Sy-QEn-HXW6H2VM:-KYu7Sy-QEn-HXW6H2VN:b-io88vp) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/00ef8982c080084c45c6937211f997678026ca6f/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsMapViewController.m#L125)):*
> ```objc
> }
> 
> -(NSString *)descriptionOfFinalsInBuilding:(CampusBuilding *)building {
> ```

Let's get some comments in this function, but it looks good :)

---

*[byuSuite/Classes/Basic Data Access/ByuFinalsSchedule/ByuFinalsScheduleRequest.h, line 26 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KYtymOU2BTsvJsbBohy:-KYtymOU2BTsvJsbBohz:bhmo0t1) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/00ef8982c080084c45c6937211f997678026ca6f/byuSuite/Classes/Basic Data Access/ByuFinalsSchedule/ByuFinalsScheduleRequest.h#L26)):*
> ```objc
> @interface ByuFinalsScheduleRequest : NSObject
> 
> @property (nonatomic, weak) id<ByuFinalsScheduleRequestDelegate> delegate;
> ```

It looks like we haven't actually refactored out the old stuff yet... Is this still being used?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50)*
<!-- Sent from Reviewable.io -->

romrell4 commented:
Review status: all files reviewed at latest revision, 20 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KYu81jjtC7y1uLT0i96:-KYu81jjtC7y1uLT0i97:bvq3ytf) (no related file):*
The callouts aren't automatically loading up for me

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50)*
<!-- Sent from Reviewable.io -->

ecbrady commented:
Review status: 12 of 14 files reviewed at latest revision, 20 unresolved discussions.

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 29 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KYtz6VChBJRbXQJoBeR:-KYzoP6N5h1SvzXfmbSA:b-896fix) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/00ef8982c080084c45c6937211f997678026ca6f/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L29)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

Can't this just be known if finals != nil?
</blockquote></details>

Done.

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 34 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KYtzCn0q8y_o8z80Gcw:-KYzoPV9l680e04-8SdE:b-896fix) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/00ef8982c080084c45c6937211f997678026ca6f/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L34)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

See comment above
</blockquote></details>

Done.

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 54 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KYtzW3LWq36H2SkTcJy:-KYzoRieu84daE7uPl8x:b-2rn9vn) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/00ef8982c080084c45c6937211f997678026ca6f/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L54)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

What is this if doing? Is this if we come back to it?
</blockquote></details>

Yeah. Since we don't want it appearing in the Map View we remove it there but then when they come back to the list view, we have to put it back. I'll add a comment explaining that.
Thanks for pointing this out, I realized it's redundant to set it both here and in the createSemesterButton so I removed it from the latter.

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 61 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KYtzc-wHziPwaMLJl86:-KYzpjGwIjsehyv5K6x_:b-896fix) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/00ef8982c080084c45c6937211f997678026ca6f/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L61)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

We shouldn't need the second one. The first one will go away when we use a Client
</blockquote></details>

Done.

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 76 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KYtzwMrXaeIAOtGZ6gb:-KYzpuPILF-5dFOWIGIW:b-896fix) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/00ef8982c080084c45c6937211f997678026ca6f/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L76)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

Put a comment here about why you're not letting them switch tabs
</blockquote></details>

Done.

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 83 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KYu--wTGRsYL6UyOswA:-KYzqjuVQu2-gQ5MCWM7:b-896fix) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/00ef8982c080084c45c6937211f997678026ca6f/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L83)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

Use ternary here
</blockquote></details>

Done.

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 104 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KYu-RUh-QmoM9tiZ3Ls:-KYzqtDKjlIU0PPPgY7M:babwjp9) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/00ef8982c080084c45c6937211f997678026ca6f/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L104)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

Are the two of these more different that just a nil detailsTextLabel?
</blockquote></details>

Yes. Clickability is only available for the first one.

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 124 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KYu-g11zPOyGUnOQjKF:-KYzrGAnFh58mynhEeoy:b-896fix) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/00ef8982c080084c45c6937211f997678026ca6f/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L124)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

Let's move this up next to the number of sections
</blockquote></details>

Done.

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 125 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KYu-v9_eSoRsXtIXEGe:-KYzrQw2oxzSGobx5ZGY:b-4yxxp8) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/00ef8982c080084c45c6937211f997678026ca6f/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L125)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

I don't think this logic is correct... If self.finals.count == 0, there will still be two sections from what it looks like....
</blockquote></details>

Currently it keeps the first section if there are no finals but we do not want the title since there are no finals in the section. That's the reason for the if else if and else (to answer the next comment). 

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 135 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KYu0EPtsDJw7q-NpDyW:-KYzs4x60Wk7p_CG5uTL:b-xoyxhk) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/00ef8982c080084c45c6937211f997678026ca6f/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L135)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

If we can do this when the view returns, that would be pretty cool :)
</blockquote></details>

It doesn't affect the animation when we've clicked on it (it stays selected until the map view has loaded). Once in the map view, there is no way for them to "go back" except to click on the List tab. If they had done some further exploring for other finals in the map view, I think it would be odd then to come back to the list view and see a final selected that isn't the last final they were looking at, if that makes sense. 

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 222 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KYu0qUnZ8t0tlm-jQPq:-KYztGw7fCNkn8hJFZCY:b-c6hndu) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/00ef8982c080084c45c6937211f997678026ca6f/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L222)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

This actually seems more like a 
</blockquote></details>

Like a what? :P  

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 225 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KYu3mscVwi2N6o75i67:-KYzuVNMBC-fapPttyVu:b-f14gs0) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/00ef8982c080084c45c6937211f997678026ca6f/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L225)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

Is this loading the buildings and final everytime the tab is selected? I don't think that would be desirable
</blockquote></details>

No it's not. This method is only called after buildings are loaded or after finals are loaded. Buildings are now only loaded once (initiated from the viewDidLoad) and finals are only loaded when/if the semester button changes.

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 233 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KYu40Bxj63xPvYvKmW7:-KYzuuDFcFoXmiPEzHGq:b-a4ivkd) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/00ef8982c080084c45c6937211f997678026ca6f/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L233)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

Should this be called [separateIntoScheduledAndUnscheduledFinals:]?
</blockquote></details>

Yeah I'll change this. I just kept the original method name but had considered changing it. However, now that I'm looking at it, it is only called once from the loadDateIfReady... Should I just combine these two methods?

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsMapViewController.m, line 44 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KYu57UX3c6GIDPaS4qU:-KZ2AZQovTxEwMWFHbKG:b-896fix) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/00ef8982c080084c45c6937211f997678026ca6f/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsMapViewController.m#L44)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

I think we wrote a higherOrder function to do this. Something called findByBlock or something like that. Try to find that in NSArray+higherOrder.h, and if you have questions, ask me
</blockquote></details>

Done.

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsMapViewController.m, line 52 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KYu5Ks5uGPjTcC2FyAX:-KZ2AZwQ4z29OfwSpdu4:b-896fix) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/00ef8982c080084c45c6937211f997678026ca6f/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsMapViewController.m#L52)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

If we can't use higher order, we should at least put a break here
</blockquote></details>

Done.

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsMapViewController.m, line 125 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KYu7Sy-QEn-HXW6H2VM:-KZ2AtnLVuYNcDPNP_oA:b-896fix) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/00ef8982c080084c45c6937211f997678026ca6f/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsMapViewController.m#L125)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

Let's get some comments in this function, but it looks good :)
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50)*
<!-- Sent from Reviewable.io -->

romrell4 commented:
Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 10 unresolved discussions.

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 104 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KYu-RUh-QmoM9tiZ3Ls:-KZ2DoKzroRXSIUBP06-:b-x345el) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/00ef8982c080084c45c6937211f997678026ca6f/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L104)):*
<details><summary><i>Previously, ecbrady wrote…</i></summary><blockquote>

Yes. Clickability is only available for the first one.
</blockquote></details>

Ah. Is that set in the storyboard?

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 222 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KYu0qUnZ8t0tlm-jQPq:-KZ2E-kkqk0zqL_4MUqw:b-7pl5ba) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/00ef8982c080084c45c6937211f997678026ca6f/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L222)):*
<details><summary><i>Previously, ecbrady wrote…</i></summary><blockquote>

Like a what? :P  
</blockquote></details>

Hahaha- sorry. Can you say ADHD much? :p I think the function could use a better name... Not sure exactly what. Maybe setUpTableIfReady or setUpIfReady or dataLoaded or something. You come up with the best name you can and run it by me

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 233 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KYu40Bxj63xPvYvKmW7:-KZ2Ecu9JmY_9NucnI7c:bmgjwfb) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/00ef8982c080084c45c6937211f997678026ca6f/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L233)):*
<details><summary><i>Previously, ecbrady wrote…</i></summary><blockquote>

Yeah I'll change this. I just kept the original method name but had considered changing it. However, now that I'm looking at it, it is only called once from the loadDateIfReady... Should I just combine these two methods?
</blockquote></details>

Yes please. It's not that large of a method

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 53 at r2](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KZ2DBEkXMw2IbmuDyBi:-KZ2DBEl5uGTWXIVC-Vf:b-jxi4s2) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/f36356e3b18efce18fe459bb0df62233868a73bd/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L53)):*
> ```objc
>     [super viewDidAppear:animated];
>     
>     //Since the semester button is removed from the tabBArController's navigationItem in the mapView, we need to put it back here
> ```

type o

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 87 at r2](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KZ2DZdP7QuBLHKw4qSL:-KZ2DZdP7QuBLHKw4qSM:b-siql5v) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/f36356e3b18efce18fe459bb0df62233868a73bd/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L87)):*
> ```objc
> 
> -(NSString *)tableView:(UITableView *)tableView titleForHeaderInSection:(NSInteger)section {
>     if (section == 0 && self.finals.count != 0) {
> ```

Explain this one to me. I saw your comment, but didn't totally understand

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 224 at r2](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KZ2EYhdNAYnmHEuZA7y:-KZ2EYhdNAYnmHEuZA7z:b-jxi4s2) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/f36356e3b18efce18fe459bb0df62233868a73bd/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L224)):*
> ```objc
>             self.tableFooter.hidden = NO;
>         }
>         [self separateIntocheduledAndUnscheduledFinals:self.finals];
> ```

type o

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsMapViewController.m, line 46 at r2](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KZ2Ej3Z0VpLR0lWbqfb:-KZ2Ej3_m5AarioT3kCQ:b-2lqc95) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/f36356e3b18efce18fe459bb0df62233868a73bd/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsMapViewController.m#L46)):*
> ```objc
>             if (![annotation isKindOfClass:[MKUserLocation class]]) {
>                 CampusBuilding *currentBuilding = annotation;
>                 return ([[currentBuilding.building lowercaseString] isEqualToString:[self.selectedFinal.building lowercaseString]]);
> ```

You shouldn't need the outer parenthesis anymore

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50)*
<!-- Sent from Reviewable.io -->

ecbrady commented:
Review status: 11 of 14 files reviewed at latest revision, 10 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KYu81jjtC7y1uLT0i96:-KZ2UiJY1b2QgBfMA5HY:b-6t35s5) (no related file):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

The callouts aren't automatically loading up for me
</blockquote></details>

They won't for finals in the testing center.

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 104 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KYu-RUh-QmoM9tiZ3Ls:-KZ2LlNK0VL_8Ke-xpNe:b6pwf7x) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/00ef8982c080084c45c6937211f997678026ca6f/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L104)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

Ah. Is that set in the storyboard?
</blockquote></details>

Yes it is.

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 222 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KYu0qUnZ8t0tlm-jQPq:-KZ2UNx7P-aKGhdfgBso:b-896fix) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/00ef8982c080084c45c6937211f997678026ca6f/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L222)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

Hahaha- sorry. Can you say ADHD much? :p I think the function could use a better name... Not sure exactly what. Maybe setUpTableIfReady or setUpIfReady or dataLoaded or something. You come up with the best name you can and run it by me
</blockquote></details>

Done.

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 233 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KYu40Bxj63xPvYvKmW7:-KZ2OfNlAqjMH4TzvnMb:b-896fix) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/00ef8982c080084c45c6937211f997678026ca6f/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L233)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

Yes please. It's not that large of a method
</blockquote></details>

Done.

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 53 at r2](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KZ2DBEkXMw2IbmuDyBi:-KZ2M7RDiMF7OLA0AlAT:bs7wumb) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/f36356e3b18efce18fe459bb0df62233868a73bd/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L53)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

type o
</blockquote></details>

Done.

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 87 at r2](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KZ2DZdP7QuBLHKw4qSL:-KZ2NDgU9OcPk0KsJfe9:b-896fix) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/f36356e3b18efce18fe459bb0df62233868a73bd/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L87)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

Explain this one to me. I saw your comment, but didn't totally understand
</blockquote></details>

Done.

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 224 at r2](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KZ2EYhdNAYnmHEuZA7y:-KZ2Of3B58frLy0QYU0d:b-896fix) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/f36356e3b18efce18fe459bb0df62233868a73bd/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L224)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

type o
</blockquote></details>

Done.

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsMapViewController.m, line 46 at r2](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KZ2Ej3Z0VpLR0lWbqfb:-KZ2UUGUBstWuyCFiq7x:b-896fix) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/f36356e3b18efce18fe459bb0df62233868a73bd/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsMapViewController.m#L46)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

You shouldn't need the outer parenthesis anymore
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50)*
<!-- Sent from Reviewable.io -->

romrell4 commented:
Reviewed 3 of 3 files at r3.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 86 at r3](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KZ2XOYGfsSkuMWKcMVc:-KZ2XOYGfsSkuMWKcMVd:b-3hrywc) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/4f2440f0e7ca45c5f5d887bdca7727374bebb746/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L86)):*
> ```objc
>         return 0;
>     } else {
>         return (self.unscheduledFinals.count == 0) ? 1 : 2;
> ```

Now that we have multiple, let's remove the ternary.

```objective-c
int sections = 0;
if (self.finals.count != 0) {
  sections++;
}
if (self.unscheduledFinals.count != 0) {
  sections++;
}
return sections;
```

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 91 at r3](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KZ2XuuPbakhrYUFBNjQ:-KZ2XuuPbakhrYUFBNjR:b-dmnhod) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/4f2440f0e7ca45c5f5d887bdca7727374bebb746/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L91)):*
> ```objc
> 
> -(NSString *)tableView:(UITableView *)tableView titleForHeaderInSection:(NSInteger)section {
>     if (section == 0) {
> ```

This will now change to
```objective-c
int sections = [self numberOfSectionsInTableView:tableView];
if (section == 0 && self.scheduledFinals.count != 0) {
  return @"  Scheduled Finals";
} else {
  return @"  Unscheduled Finals";
}
```

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 136 at r3](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KZ2_7mGZ5gxn3-6w_n-:-KZ2_7mHlZkSS2tv1Sbp:blgxwhs) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/4f2440f0e7ca45c5f5d887bdca7727374bebb746/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L136)):*
> ```objc
> -(void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
>     [self.tableView deselectRowAtIndexPath:indexPath animated:YES];
>     if (indexPath.section == 0) {
> ```

Don't need to check for the section anymore. But put a comment saying that only the scheduleFinals will be clickable

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50)*
<!-- Sent from Reviewable.io -->

romrell4 commented:
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 86 at r3](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KZ2XOYGfsSkuMWKcMVc:-KZ2bv196DsXykKACs2I:bpq3syo) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/4f2440f0e7ca45c5f5d887bdca7727374bebb746/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L86)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

Now that we have multiple, let's remove the ternary.

```objective-c
int sections = 0;
if (self.finals.count != 0) {
  sections++;
}
if (self.unscheduledFinals.count != 0) {
  sections++;
}
return sections;
```
</blockquote></details>

Sorry,

```objective-c
int sections = 0;
if (self.testingCenterFinals.count != 0) {
  sections++;
}
if (self.scheduledFinals.count != 0) {
  sections++;
}
if (self.unscheduledFinals.count != 0) {
  sections++;
}
return sections;
```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50)*
<!-- Sent from Reviewable.io -->

romrell4 commented:
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 91 at r3](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KZ2XuuPbakhrYUFBNjQ:-KZ2cA2UzIROnEOaLXHl:b-c4jaed) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/4f2440f0e7ca45c5f5d887bdca7727374bebb746/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L91)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

This will now change to
```objective-c
int sections = [self numberOfSectionsInTableView:tableView];
if (section == 0 && self.scheduledFinals.count != 0) {
  return @"  Scheduled Finals";
} else {
  return @"  Unscheduled Finals";
}
```
</blockquote></details>

Sorry, 
```objective-c
int sections = [self numberOfSectionsInTableView:tableView];
if (section == 0) {
  if (self.testingCenterFinals.count != 0) {
    return @"  Testing Center Administered Finals";
  } else if (self.
}
if (section == 0 && self.scheduledFinals.count != 0) {
  return @"  Scheduled Finals";
} else {
  return @"  Unscheduled Finals";
}
```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50)*
<!-- Sent from Reviewable.io -->

romrell4 commented:
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 91 at r3](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KZ2XuuPbakhrYUFBNjQ:-KZ2cU6OrmzxKsqYmXBK:b-677oru) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/4f2440f0e7ca45c5f5d887bdca7727374bebb746/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L91)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

Sorry, 
```objective-c
int sections = [self numberOfSectionsInTableView:tableView];
if (section == 0) {
  if (self.testingCenterFinals.count != 0) {
    return @"  Testing Center Administered Finals";
  } else if (self.
}
if (section == 0 && self.scheduledFinals.count != 0) {
  return @"  Scheduled Finals";
} else {
  return @"  Unscheduled Finals";
}
```
</blockquote></details>

Sorry again,
```objective c
int sections = [self numberOfSectionsInTableView:tableView];
if (section == 0) {
  if (self.testingCenterFinals.count > 0) {
    return @"  Testing Center Administered Finals";
  } else if (self.scheduledFinals.count > 0) {
    return @"  Scheduled Finals";
  } else {
    return @"  Unscheduled Finals";
  }
} else if (section == 1) {
  if (self.testingCenterFinals.count > 0) {
    if (self.scheduledFinals.count > 0) {
      return @"  Scheduled Finals";
    } else {
      return @"  Unscheduled Finals";
    }
  } else if (self.scheduledFinals.count > 0) {
    return @"  Unscheduled Finals";
  }
} else {
  return @"  Unscheduled Finals";
}
```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50)*
<!-- Sent from Reviewable.io -->

ecbrady commented:
Review status: 12 of 14 files reviewed at latest revision, 5 unresolved discussions.

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 86 at r3](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KZ2XOYGfsSkuMWKcMVc:-KZ2dmATx7YIOIVx2cY7:b-896fix) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/4f2440f0e7ca45c5f5d887bdca7727374bebb746/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L86)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

Sorry,

```objective-c
int sections = 0;
if (self.testingCenterFinals.count != 0) {
  sections++;
}
if (self.scheduledFinals.count != 0) {
  sections++;
}
if (self.unscheduledFinals.count != 0) {
  sections++;
}
return sections;
```
</blockquote></details>

Done.

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 91 at r3](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KZ2XuuPbakhrYUFBNjQ:-KZ2eo98eQkOGWB1TRQg:b-896fix) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/4f2440f0e7ca45c5f5d887bdca7727374bebb746/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L91)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

Sorry again,
```objective c
int sections = [self numberOfSectionsInTableView:tableView];
if (section == 0) {
  if (self.testingCenterFinals.count > 0) {
    return @"  Testing Center Administered Finals";
  } else if (self.scheduledFinals.count > 0) {
    return @"  Scheduled Finals";
  } else {
    return @"  Unscheduled Finals";
  }
} else if (section == 1) {
  if (self.testingCenterFinals.count > 0) {
    if (self.scheduledFinals.count > 0) {
      return @"  Scheduled Finals";
    } else {
      return @"  Unscheduled Finals";
    }
  } else if (self.scheduledFinals.count > 0) {
    return @"  Unscheduled Finals";
  }
} else {
  return @"  Unscheduled Finals";
}
```
</blockquote></details>

Done.

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 136 at r3](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KZ2_7mGZ5gxn3-6w_n-:-KZ2fFVrVc3C8YHUBTKR:b-896fix) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/4f2440f0e7ca45c5f5d887bdca7727374bebb746/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L136)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

Don't need to check for the section anymore. But put a comment saying that only the scheduleFinals will be clickable
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50)*
<!-- Sent from Reviewable.io -->

romrell4 commented:
Reviewed 2 of 2 files at r4.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50)*
<!-- Sent from Reviewable.io -->

ecbrady commented:
Review status: 8 of 20 files reviewed at latest revision, 3 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KZ3K0w747pReDNG8XvV:-KZ3K0w747pReDNG8XvW:b-8i1nnp) (no related file):*
Don't forget to look into renaming this feature.

---

*[byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m, line 28 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KYtz1A-KBhEGbKsf2Uf:-KZ3P-4I04MWd3I2yVUe:b-896fix) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/00ef8982c080084c45c6937211f997678026ca6f/byuSuite/Apps/MapMyFinals/controller/MapMyFinalsListViewController.m#L28)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

We need to replace this with a Client. Maybe that wasn't part of this phase, but we'll have to do it before we're done with this feature
</blockquote></details>

Done.

---

*[byuSuite/Classes/Basic Data Access/ByuFinalsSchedule/ByuFinalsScheduleRequest.h, line 26 at r1](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KYtymOU2BTsvJsbBohy:-KZ3P0AgVzVrDJs2T8iR:b-896fix) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/00ef8982c080084c45c6937211f997678026ca6f/byuSuite/Classes/Basic Data Access/ByuFinalsSchedule/ByuFinalsScheduleRequest.h#L26)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

It looks like we haven't actually refactored out the old stuff yet... Is this still being used?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50)*
<!-- Sent from Reviewable.io -->

romrell4 commented:
Reviewed 12 of 12 files at r5.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KZ3K0w747pReDNG8XvV:-KZ3UzxDLjLeJjTpXChw:b-sc2ixq) (no related file):*
<details><summary><i>Previously, ecbrady wrote…</i></summary><blockquote>

Don't forget to look into renaming this feature.
</blockquote></details>

Yes. Don't forget about it :)

---

*[byuSuite/Apps/MapMyFinals/model/MapMyFinalsClient.m, line 31 at r5](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KZ3URULBaLaSWrV8jFc:-KZ3URULBaLaSWrV8jFd:bkehufa) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/6daa26dc2115d1b27fb8902b7828f20a65124889/byuSuite/Apps/MapMyFinals/model/MapMyFinalsClient.m#L31)):*
> ```objc
>             NSArray *data = [[response getDataJson] nullSafeObjectForKeyPath:@"ExamStudentScheduleService.response.schedule_table"];
>             
>             NSArray *finalDicts = [data filterWithBlock:^BOOL(NSDictionary *dict) {
> ```

What's this guy doing? Definitely throw a comment above this line. Also, you probably don't have to use nullSafeObjectForKey here. Just use dict[@"is_term_header"]. Or even easier, do the transform first, and then the filter will just be `return ![final.isTermHeader isEqualToString[@"Y"]`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50)*
<!-- Sent from Reviewable.io -->

romrell4 commented:
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KZ3K0w747pReDNG8XvV:-KZ3VbRGmfn35iXcdN19:b-p6hiii) (no related file):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

Yes. Don't forget about it :)
</blockquote></details>

I slacked Joann. I'll let you know when I get a response from her

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50)*
<!-- Sent from Reviewable.io -->

ecbrady commented:
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[byuSuite/Apps/MapMyFinals/model/MapMyFinalsClient.m, line 31 at r5](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KZ3URULBaLaSWrV8jFc:-KZ3_qhAmoeAIr4GY26E:b-896fix) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/6daa26dc2115d1b27fb8902b7828f20a65124889/byuSuite/Apps/MapMyFinals/model/MapMyFinalsClient.m#L31)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

What's this guy doing? Definitely throw a comment above this line. Also, you probably don't have to use nullSafeObjectForKey here. Just use dict[@"is_term_header"]. Or even easier, do the transform first, and then the filter will just be `return ![final.isTermHeader isEqualToString[@"Y"]`
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50)*
<!-- Sent from Reviewable.io -->

romrell4 commented:
Reviewed 1 of 1 files at r6.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50)*
<!-- Sent from Reviewable.io -->

ecbrady commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[a discussion](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KZ3K0w747pReDNG8XvV:-KZ7AIzED22IsdBSj4eG:bkzj7xi) (no related file):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

I slacked Joann. I'll let you know when I get a response from her
</blockquote></details>

She says she's okay with it. Just let me know what you'd like me to rename it to.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50)*
<!-- Sent from Reviewable.io -->

romrell4 commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[a discussion](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KZ3K0w747pReDNG8XvV:-KZ9Li62Pzaex7LwsP6h:b-nxhpx) (no related file):*
<details><summary><i>Previously, ecbrady wrote…</i></summary><blockquote>

She says she's okay with it. Just let me know what you'd like me to rename it to.
</blockquote></details>

My Finals Schedule

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50)*
<!-- Sent from Reviewable.io -->

romrell4 commented:
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KZ9LkU_r5dVINvFw7rm:-KZ9LkU_r5dVINvFw7rn:by8nvr5) (no related file):*
Run the formatter

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50)*
<!-- Sent from Reviewable.io -->

ecbrady commented:
Review status: 9 of 20 files reviewed at latest revision, 2 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KZ3K0w747pReDNG8XvV:-KZNCaVvdBitnQ9kdaxF:b-896fix) (no related file):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

My Finals Schedule
</blockquote></details>

Done.

---

*[a discussion](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KZ9LkU_r5dVINvFw7rm:-KZNCwqDW2avSL37lE_4:b-896fix) (no related file):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

Run the formatter
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50)*
<!-- Sent from Reviewable.io -->

romrell4 commented:
Reviewed 1 of 20 files at r1, 20 of 20 files at r7.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KZNUCm9ps94E3O6yTKR:-KZNUCm9ps94E3O6yTKS:bysiybn) (no related file):*
Did we remove the old objects?

---

*[byuSuite/Apps/MyFinalsSchedule/controller/MyFinalsScheduleListViewController.m, line 31 at r7](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KZNTmZ00jUx9cqFjMEX:-KZNTmZ00jUx9cqFjMEY:bwfrh9l) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/cc1d88c20a31529af106c3a6f23ccd540bf02244/byuSuite/Apps/MyFinalsSchedule/controller/MyFinalsScheduleListViewController.m#L31)):*
> ```objc
> @property (nonatomic) SemesterButton *semButton;
> @property (nonatomic) RegYearTerm *yearTerm;
> @property (nonatomic) NSArray<MyFinalsScheduleFinalExam *> *finals;
> ```

Was there a reason we renamed this one? It seemed fine before. I'm okay if you have a good reason though

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50)*
<!-- Sent from Reviewable.io -->

ecbrady commented:
Review status: 12 of 20 files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

*[a discussion](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KZNUCm9ps94E3O6yTKR:-KZNXs0VYi3MAaQl41nN:b-jity52) (no related file):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

Did we remove the old objects?
</blockquote></details>

Yes I removed the ByuFinalsSchedule from Basic Data Access.

---

*[byuSuite/Apps/MyFinalsSchedule/controller/MyFinalsScheduleListViewController.m, line 31 at r7](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50#-KZNTmZ00jUx9cqFjMEX:-KZNX3CEsY0EzJYYAIYH:bm3x4bf) ([raw file](https://github.com/byu-oit-appdev/ios-byusuite/blob/cc1d88c20a31529af106c3a6f23ccd540bf02244/byuSuite/Apps/MyFinalsSchedule/controller/MyFinalsScheduleListViewController.m#L31)):*
<details><summary><i>Previously, romrell4 (Eric Romrell) wrote…</i></summary><blockquote>

Was there a reason we renamed this one? It seemed fine before. I'm okay if you have a good reason though
</blockquote></details>

No I liked it better that way too. I'll change it back.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50)*
<!-- Sent from Reviewable.io -->

romrell4 commented:
Reviewed 6 of 22 files at r7, 9 of 10 files at r8.
Review status: 19 of 20 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50)*
<!-- Sent from Reviewable.io -->

romrell4 commented:
Reviewed 1 of 22 files at r7, 1 of 10 files at r8.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/byu-oit-appdev/ios-byusuite/50)*
<!-- Sent from Reviewable.io -->
